lxd/storage_volumes.go

```diff
@@ -2727,3 +2727,84 @@ func createStoragePoolVolumeFromBackup(s *state.State, r *http.Request, requestP
 	revert.Success()
 	return operations.OperationResponse(op)
 }
+
+// getRemoteVolumeNodeInfo figures out the cluster member on which the volume with the given name is defined. If it is
+// the local cluster member it returns nil and no error. If it is another cluster member it returns a db.NodeInfo containing
+// the name and address of the remote member. If there is more than one cluster member with a matching volume name, an
+// error is returned.
+func getRemoteVolumeNodeInfo(s *state.State, poolName string, projectName string, volumeName string, volumeType int) (*db.NodeInfo, error) {
```

Comment:
New DB functions should be taking a context arg so we can pass it to queries to allow cancellation in my view.